robottelo/ui/contentviews.py

```diff
@@ -450,6 +462,45 @@ def add_remove_errata_to_filter(self, cv_name, filter_name,
         else:
             self.click(locators['contentviews.remove_errata'])
 
+    def edit_erratum_date_range_filter(
+            self, cv_name, filter_name, errata_types=None, date_type=None,
+            start_date=None, end_date=None, open_filter=True):
+        """Edit Erratum Date Range Filter"""
+        allowed_errata_types = FILTER_ERRATA_TYPE.values()
+        allowed_date_types = FILTER_ERRATA_DATE.values()
+        erratum_type_locator = locators['contentviews.erratum_type_checkbox']
+        date_type_locator = locators['contentviews.erratum_date_type']
+        if open_filter:
+            self.go_to_filter_page(cv_name, filter_name)
+        if errata_types is not None:
+            if errata_types:
+                # because of the behaviour of the UI to check and disable
+                # the last element if the others are unchecked,
+                # will check all the elements first,
+                # after then uncheck the not selected ones
+                for errata_type in allowed_errata_types:
+                    self.assign_value(erratum_type_locator % errata_type, True)
+                for errata_type in allowed_errata_types:
+                    if errata_type not in errata_types:
+                        self.assign_value(
+                            erratum_type_locator % errata_type, False)
```

Comment:
NACK, this is basically wrong. 2 loops for just checking checkboxes is not ok.
Think of this:
```python
for errata in allowed_errata_types:
    self.assign_value(
        locators['contentviews.erratum_type_checkbox'] % checkbox,
        errata in errata_types
    )
```
So instead of firstly checking everything on and then unchecking what's not supposed to be checked you will set correct state for each checkbox one by one